### PR TITLE
List nodes

### PR DESCRIPTION
**Pull request by matsuuran**
PR for #606 

first real pull so hopefully everything is in order

### CODE REVIEW

**Comment by shwsun** [comment]
I tested it in devhaas and it looks fine, so +1.

File: docs/apidesc.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ one exception is NICs, where the label is unique only on a per-node basis.
 
     # Query interface
 
-    list_free_nodes
+    list_node                  <is_free>
```
Comment (knikolla) [inline docs/apidesc.md]: list_nodes

File: haas/api.py [hunk 2]
```diff
@@ -796,20 +796,22 @@ def port_detach_nic(switch, port):
     port.nic = None
     db.session.commit()
 
-
-@rest_call('GET', '/free_nodes', Schema({}))
-def list_free_nodes():
-    """List all nodes not in any project.
+@rest_call('GET', '/node/<is_free>', Schema({'is_free' : basestring}))
```
Comment (knikolla) [inline haas/api.py]: Should this be `/node` or `/nodes`? The API description says `nodes` but here you are registering it to `node`.

Comment (matsuuran) [inline haas/api.py]: I think it should be /node so if for whatever reason we wanted to grab nodes by name or whatever we could do so with /node/<name>.  I'm not terribly committed to it so if you feel strongly about it and have reason, I can change it.

File: docs/rest_api.md [hunk 3]
```diff
@@ -329,11 +329,12 @@ Authorization requirements:
 
 * Access to the project to which `<node>` is assigned (if any) or administrative access.
 
-### list_free_nodes
+### list_nodes
 
-`GET /free_nodes`
+`GET /nodes/<is_free>`
```
Comment (SahilTikale) [inline docs/rest_api.md]: make this consistent with the API call. `node/<is_free>`

File: haas/api.py (hunk 2 above)
Comment (knikolla) [inline haas/api.py]: I don't feel strongly about either, but for some reason list_projects is registered to `GET /projects`. https://github.com/CCI-MOC/hil/blob/master/docs/rest_api.md#list_projects so I'd prefer to be consistent. But we should discuss about this.

Comment (SahilTikale) [inline haas/api.py]: I checked online and people use both plural and singular forms equally. I prefer idea of a resource name without getting into "English" of its plurality. As it is, many plural words do not have well defined patterns. child - children, foot -feet, sheep -sheep. So to be consistent and avoid any such problems in the future, lets stick with the singular form of resources. Whatever api is plural should be changed as we go.

**Comment by matsuuran** [comment]
is everyone cool with this now?

**Comment by SahilTikale** [comment]
This works. merging and closing it.

**Pull request merged**